lib/network.js

```diff
@@ -13,7 +13,9 @@ module.exports = function (archive, opts, cb) {
     stream: opts.stream
   }, opts)
   var swarm = disc(swarmDefaults(swarmOpts))
-  swarm.once('error', function () {
+  swarm.once('error', function (err) {
+    // Don't magically use any port if the users has requested a specific port
+    swarm.emit('errorport', err)
```

Comment:
Can also we call the callback, `cb`?

Reply:
mmm, looking at that file - no idea, I don't know what it does, or how / when it should be called :/

Reply:
ah, ok, that changes the if answer too - awesome, updating.

Reply:
> mmm, looking at that file - no idea, I don't know what it does, or how / when it should be called :/

The callback can be used to know if there are any network connections (its called back after first round of discovery). But we can pass the error back as the first argument as well.